app/assets/src/components/views/bulk_download/ChooseStep.jsx

```diff
@@ -49,83 +88,154 @@ class ChooseStep extends React.Component {
             value: "all",
+            customNode: (
+              <div className={cs.taxonWithReadsOption}>
+                <div className={cs.taxonName}>All taxon</div>
+                <div className={cs.fill} />
+                <div className={cs.sampleCount}>{selectedSampleIds.size}</div>
+              </div>
+            ),
           },
+          // Note: BareDropdown with search prioritizes prefix matches, so the final ordering of options
+          // might not be the one provided here.
+          ...(this.sortTaxonWithReadsOptions(taxonWithReadsOptions) || []),
         ];
 
-        return (
-          <div className={cs.field} key={field.type}>
-            <div className={cs.label}>{field.display_name}:</div>
-            <Dropdown
-              fluid
-              options={dropdownOptions}
-              onChange={(value, displayName) =>
-                onFieldSelect(downloadType.type, field.type, value, displayName)
-              }
-              value={selectedField}
-            />
+        placeholder = "Select taxa";
+        menuLabel = "Select taxa";
+        showNoResultsMessage = true;
+        search = true;
+        onFilterChange = this.handleTaxonWithReadsSelectFilterChange;
+        className = cs.taxonWithReadsDropdown;
+        loadingSearchOptions = loadingTaxonWithReadsOptions;
+
+        optionsHeader = (
+          <div className={cs.taxonWithReadsOptionsHeader}>
+            <div className={cs.header}>Taxa</div>
+            <div className={cs.fill} />
+            <div className={cs.header}>Samples</div>
           </div>
         );
-      // TODO(mark): Consolidate this with the above fields once the set of fields is stable.
+        break;
       case "background":
-        if (fieldOptions.backgrounds) {
-          dropdownOptions = fieldOptions.backgrounds.map(background => ({
-            text: background.name,
-            value: background.id,
-          }));
-        } else {
-          loadingOptions = true;
-        }
-
-        return (
-          <div className={cs.field} key={field.type}>
-            <div className={cs.label}>{field.display_name}:</div>
-            <Dropdown
-              fluid
-              disabled={loadingOptions}
-              placeholder={loadingOptions && "Loading..."}
-              options={dropdownOptions || []}
-              onChange={(value, displayName) =>
-                onFieldSelect(downloadType.type, field.type, value, displayName)
-              }
-              value={selectedField}
-            />
-          </div>
-        );
-      default:
-        return null;
+        dropdownOptions = backgroundOptions || [];
+        placeholder = backgroundOptions ? "Select background" : "Loading...";
+        break;
+    }
+
+    if (!dropdownOptions) {
+      return null;
     }
+
+    return (
+      <div className={cs.field} key={field.type}>
+        <div className={cs.label}>{field.display_name}:</div>
+        <Dropdown
```

Comment:
It just does not feel like this should be using the Dropdown directly... So none of the other higher level components makes sense here?
It seems like setting up this dropdown was too complex...

Reply:
Good point. `<LiveSearchPopBox>` is the one with the closest set of features, in that it also fetches data with a debounce. But that one is also quite different, because it allows users to pick an option OR type whatever they want, whereas in my case the user is required to pick an option. There's no clear alternative dropdown to use.

We should definitely think about refactoring these dropdowns soon.

Reply:
One concrete thing we can do is refactor the ability to load options asynchronously into a wrapper component or something like that.

Reply:
Yeah, i guess i left this comment more as an extra note for future discussion.